mne/beamformer/_dics.py

```diff
@@ -421,6 +423,97 @@ def apply_dics_epochs(epochs, filters, return_generator=False, verbose=None):
     return stcs
 
 
+@verbose
+def apply_dics_tfr(epochs, filters, freqs, method='morlet', average=False,
+                   n_csd=256, n_overlap=0, verbose=None, **csd_kwargs):
+    """Apply Dynamic Imaging of Coherent Sources (DICS) beamformer weights.
+
+    Apply Dynamic Imaging of Coherent Sources (DICS) beamformer weights
+    on single trial time-frequency data.
+
+    Parameters
+    ----------
+    epochs : Epochs
+        Single trial epochs.
+    filters : instance of Beamformer
+        DICS spatial filter (beamformer weights)
+        Filter weights returned from :func:`make_dics`.
+    freqs : ndarray, shape (n_freqs,)
+        The frequencies over which to compute cross-spectral density.
+    method : 'morlet' | 'fourier' | 'multitaper'
+        The cross-spectral density method to use.
+    average : bool
+        Whether to average across epochs. Default False.
```

Comment:
If we follow `apply_` tradition, epochs and averaged data should be two different functions. I would concentrate on averaged data first, to keep it simple - that is also what you need for your visualization. If folks want to browse epochs, they'd usually do that in sensor space, I reckon. I would concentrate on averages first, and if there is time add the rest later.

Reply:
I agree that that is easier but I took on this GSoC because I wanted to do single-trial resolution for my own data analysis purposes so it would be nice to allow for the more general case.